pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -256,8 +257,9 @@ private InstanceResponseBlock executeInternal(ServerQueryRequest queryRequest, E
 
     InstanceResponseBlock instanceResponse = null;
     try {
-      instanceResponse = executeInternal(indexSegments, queryContext, timerContext, executorService, streamer,
-          queryRequest.isEnableStreaming());
+      instanceResponse =
+          executeInternal(tableDataManager, indexSegments, queryContext, timerContext, executorService, streamer,
+              queryRequest.isEnableStreaming());
```

Comment:
nit: can we pass List<SegmentContext> itself instead of tableDataManager?

Reply:
inside executeInternal() the list of indexSegments is pruned firstly, so passing tableDataManager into executeInternal() to call the getSegmentContexts() method after pruning just for the remaining indexSegments.